<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BEAST2 (&lt;https://www.beast2.org&gt;) is a widely used
  Bayesian phylogenetic tool, that uses DNA/RNA/protein data
  and many model priors to create a posterior of jointly estimated 
  phylogenies and parameters.
  BEAUti 2 (which is part of BEAST2) is a GUI tool 
  that allows users to specify the many possible setups
  and generates the XML file BEAST2 needs to run.
  This package provides a way to create BEAST2 input
  files without active user input, but using
  R function calls instead.">
<title>BEAUti from R • beautier</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="BEAUti from R">
<meta property="og:description" content="BEAST2 (&lt;https://www.beast2.org&gt;) is a widely used
  Bayesian phylogenetic tool, that uses DNA/RNA/protein data
  and many model priors to create a posterior of jointly estimated 
  phylogenies and parameters.
  BEAUti 2 (which is part of BEAST2) is a GUI tool 
  that allows users to specify the many possible setups
  and generates the XML file BEAST2 needs to run.
  This package provides a way to create BEAST2 input
  files without active user input, but using
  R function calls instead.">
<meta property="og:image" content="https://docs.ropensci.org/beautier/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="index.html">beautier</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.6.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/demo.html">beautier demo</a>
    <a class="dropdown-item" href="articles/examples.html">Examples</a>
    <a class="dropdown-item" href="articles/inference_models.html">Inference Models</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/beautier/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="beautier">beautier<a class="anchor" aria-label="anchor" href="#beautier"></a>
</h1>
</div>
<p><a href="https://github.com/ropensci/software-review/issues/209" class="external-link"><img src="https://badges.ropensci.org/209_status.svg" alt="Peer Review Status"></a> <a href="https://cran.r-project.org/package=beautier" class="external-link"><img src="http://www.r-pkg.org/badges/version/beautier" alt="CRAN_Status_Badge"></a> <a href="https://CRAN.R-project.org/package=beautier" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/grand-total/beautier"></a> <a href="https://CRAN.R-project.org/package=beautier" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/beautier"></a> <a href="https://zenodo.org/badge/latestdoi/53443354" class="external-link"><img src="https://zenodo.org/badge/53443354.svg" alt="DOI"></a></p>
<table class="table">
<thead><tr class="header">
<th>Branch</th>
<th><a href="https://github.com/ropensci/beautier/actions" class="external-link"><img src="reference/figures/GitHubActions.png" alt="GitHub Actions logo"></a></th>
<th><a href="https://about.codecov.io/" class="external-link"><img src="reference/figures/Codecov.png" alt="Codecov logo"></a></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>master</code></td>
<td><img src="https://github.com/ropensci/beautier/workflows/R-CMD-check/badge.svg?branch=master" alt="R-CMD-check"></td>
<td><a href="https://app.codecov.io/github/ropensci/beautier/branch/master" class="external-link"><img src="https://codecov.io/github/ropensci/beautier/coverage.svg?branch=master" alt="codecov.io"></a></td>
</tr>
<tr class="even">
<td><code>develop</code></td>
<td><img src="https://github.com/ropensci/beautier/workflows/R-CMD-check/badge.svg?branch=develop" alt="R-CMD-check"></td>
<td><a href="https://app.codecov.io/github/ropensci/beautier/branch/develop" class="external-link"><img src="https://codecov.io/github/ropensci/beautier/coverage.svg?branch=develop" alt="codecov.io"></a></td>
</tr>
</tbody>
</table>
<p><code>beautier</code> is <code>BEAUti</code> for R.</p>
<div class="figure">
<img src="reference/figures/beautier_logo.png" alt=""><p class="caption">beautier logo</p>
</div>
<p>The purpose of <code>beautier</code> is to create <a href="inst/extdata/2_4.xml">a valid BEAST2 XML input file</a> from a n inference model. In this way, a scientific pipeline using <code>BEAST2</code> can be fully scripted, instead of using <code>BEAUti</code>’s GUI.</p>
<p><code>beautier</code> is part of the <a href="https://github.com/ropensci/babette" class="external-link"><code>babette</code></a> package suite:</p>
<ul>
<li>
<a href="https://github.com/ropensci/beautier" class="external-link"><code>beautier</code></a> create a BEAST2 input (<code>.xml</code>) file from an inference model.</li>
<li>
<a href="https://github.com/richelbilderbeek/tiebeaur" class="external-link"><code>tiebeaur</code></a> creates an inference model from a BEAST2 input (<code>.xml</code>) file <span class="emoji" data-emoji="warning">⚠️</span> experimental <span class="emoji" data-emoji="warning">⚠️</span>
</li>
<li>
<a href="https://github.com/ropensci/beastier" class="external-link"><code>beastier</code></a> runs BEAST2</li>
<li>
<a href="https://github.com/ropensci/tracerer" class="external-link"><code>tracerer</code></a> pastes BEAST2 output (<code>.log</code>, <code>.trees</code>, etc) files.</li>
<li>
<a href="https://github.com/ropensci/mauricer" class="external-link"><code>mauricer</code></a> install BEAST2 packages</li>
</ul>
<p>Related R packages:</p>
<ul>
<li>
<a href="https://github.com/richelbilderbeek/beautier_on_windows" class="external-link"><code>beautier_on_windows</code></a>: verifies <code>beautier</code> builds on Windows</li>
<li>
<a href="https://github.com/ropensci/lumier" class="external-link"><code>lumier</code></a>: Shiny app to help create the function call needed</li>
</ul>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>See <a href="doc/examples.html">examples</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>beautier</code> can be installed:</p>
<ul>
<li>Latest CRAN version: CRAN</li>
<li>Latest stable version: GitHub, <code>master</code> branch</li>
<li>Bleeding-edge version: GitHub, <code>develop</code> branch</li>
</ul>
<div class="section level3">
<h3 id="cran">CRAN<a class="anchor" aria-label="anchor" href="#cran"></a>
</h3>
<p>For the latest CRAN version:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"beautier"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="github-master-branch">GitHub, <code>master</code> branch<a class="anchor" aria-label="anchor" href="#github-master-branch"></a>
</h3>
<p>For the latest stable version:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/beautier"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="github-develop-branch">GitHub, <code>develop</code> branch<a class="anchor" aria-label="anchor" href="#github-develop-branch"></a>
</h3>
<p>For the bleeding-edge version:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/beautier"</span>, ref <span class="op">=</span> <span class="st">"develop"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="faq">
<a href="doc/faq.html">FAQ</a><a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<p>See <a href="doc/faq.html">FAQ</a>.</p>
</div>
<div class="section level2">
<h2 id="supported">Supported<a class="anchor" aria-label="anchor" href="#supported"></a>
</h2>
<p>This works, and the interface is unlikely to change.</p>
<ul>
<li>1 DNA alignment</li>
<li>Site models:<ul>
<li>JC69</li>
<li>HKY</li>
<li>TN93</li>
<li>GTR</li>
</ul>
</li>
<li>Clock models:<ul>
<li>Strickt</li>
<li>Relaxed log-normal</li>
</ul>
</li>
<li>Tree models:<ul>
<li>Yule</li>
<li>Birth-Death</li>
<li>Coalescent Bayesian Skyline</li>
<li>Coalescent Constant Population</li>
<li>Coalescent Exponential Population</li>
</ul>
</li>
<li>Handle missing data: simply use a dash (´-´) as a sequence in a FASTA file</li>
</ul>
</div>
<div class="section level2">
<h2 id="experimental">Experimental<a class="anchor" aria-label="anchor" href="#experimental"></a>
</h2>
<p>This works partially, and the interface may change as well.</p>
<div class="section level3">
<h3 id="tip-dating">Tip dating<a class="anchor" aria-label="anchor" href="#tip-dating"></a>
</h3>
<p>The tip dates file is a file that needs to not have column, nor row names. The columns need to be tab separated.</p>
<p>See <a href="https://github.com/ropensci/beautier/blob/master/inst/extdata/G_VII_pre2003_dates_4.txt" class="external-link">here</a> for an example, of which the first rows are shown here:</p>
<pre><code>KF767106_Indonesia_1976_VII 1976
KF767104_Indonesia_1988_VII 1988
KF767105_Indonesia_1988_VII 1988
AY288998_Indonesia_1990_VII 1990</code></pre>
<p>In the future, there probably will be a ´to_tipdates_file´ function, to create a temporary tipdates file from a table.</p>
</div>
</div>
<div class="section level2">
<h2 id="missing-featuresunsupported">Missing features/unsupported<a class="anchor" aria-label="anchor" href="#missing-featuresunsupported"></a>
</h2>
<p><code>beautier</code> cannot do everything <code>BEAUti</code> can.</p>
<p>Here are some missing or (yet) unsupported features, some are linked to an Issue:</p>
<ul>
<li><a href="https://github.com/ropensci/beautier/issues/130" class="external-link">Add offset to a distribution</a></li>
<li>Two or more DNA alignments</li>
<li>Two or more site, clock or tree models</li>
<li><a href="https://github.com/ropensci/beautier/issues/131" class="external-link">Two or more MRCA priors</a></li>
<li>Shared site, clock and/or tree models</li>
<li><a href="https://github.com/ropensci/beautier/issues/114" class="external-link">Using an amino acid alignment</a></li>
<li>Support for hyper parameters</li>
<li>Clock models<ul>
<li>Relaxed exponential</li>
<li>Random local</li>
</ul>
</li>
<li>Tree priors<ul>
<li>Calibrated Yule model</li>
<li>Coalescent Extended Bayesian Skyline</li>
<li><a href="https://github.com/ropensci/beautier/issues/133" class="external-link">Birth Death Skyline Serial</a></li>
</ul>
</li>
<li>Initialization (this is a tab that is hidden by default in <code>BEAUti</code>)</li>
</ul>
</div>
<div class="section level2">
<h2 id="there-is-a-feature-i-miss">There is a feature I miss<a class="anchor" aria-label="anchor" href="#there-is-a-feature-i-miss"></a>
</h2>
<p>See <a href="CONTRIBUTING.html">CONTRIBUTING</a>, at <code>Submitting use cases</code></p>
</div>
<div class="section level2">
<h2 id="i-want-to-collaborate">I want to collaborate<a class="anchor" aria-label="anchor" href="#i-want-to-collaborate"></a>
</h2>
<p>See <a href="CONTRIBUTING.html">CONTRIBUTING</a>, at ‘Submitting code’</p>
</div>
<div class="section level2">
<h2 id="i-think-i-have-found-a-bug">I think I have found a bug<a class="anchor" aria-label="anchor" href="#i-think-i-have-found-a-bug"></a>
</h2>
<p>See <a href="CONTRIBUTING.html">CONTRIBUTING</a>, at ‘Submitting bugs’</p>
</div>
<div class="section level2">
<h2 id="theres-something-else-i-want-to-say">There’s something else I want to say<a class="anchor" aria-label="anchor" href="#theres-something-else-i-want-to-say"></a>
</h2>
<p>Sure, just add an Issue. Or send an email.</p>
</div>
<div class="section level2">
<h2 id="external-links">External links<a class="anchor" aria-label="anchor" href="#external-links"></a>
</h2>
<ul><li><a href="https://github.com/CompEvol/beast2" class="external-link">BEAST2 GitHub</a></li></ul>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Article about <code>babette</code>:</p>
<ul><li>Bilderbeek, Richèl JC, and Rampal S. Etienne. “<code>babette</code>: BEAUti 2, BEAST 2 and Tracer for R.” Methods in Ecology and Evolution (2018). <a href="https://doi.org/10.1111/2041-210X.13032" class="external-link uri">https://doi.org/10.1111/2041-210X.13032</a>
</li></ul>
<p>FASTA files <code>anthus_aco.fas</code> and <code>anthus_nd2.fas</code> from:</p>
<ul><li>Van Els, Paul, and Heraldo V. Norambuena. “A revision of species limits in Neotropical pipits Anthus based on multilocus genetic and vocal data.” Ibis.</li></ul>
<p>FASTA file <code>G_VII_pre2003_msa.fas</code> from:</p>
<ul><li>Durr, PA; Wibowo, MH; Tabbu, CR; Asmara, W; Selleck, P; Wang, J; Broz, I; Graham, K.; Dimitrov, K and Afonso, C. (in preparation). Phylodynamics of Genotype VII Newcastle disease virus in Indonesia.</li></ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=beautier" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ropensci/beautier/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/beautier/issues" class="external-link">Report a bug</a></li>
</ul>
</div>



<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing beautier</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Richèl J.C. Bilderbeek <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1107-7049" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="r-universe">
<h2 data-toc-skip>R-universe</h2>
<ul class="list-unstyled">
<li><p><a href="https://ropensci.r-universe.dev/beautier" class="external-link"><img src="https://ropensci.r-universe.dev/badges/beautier" alt="beautier status badge"></a></p></li>
</ul>
</div>

  </aside>
</div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
