<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ beautier</title><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="https://docs.ropensci.org/beautier/logo.png"><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css"><script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript><!-- End Matomo Code --></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">beautier</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.6.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/demo.html">beautier demo</a>
    <a class="dropdown-item" href="../articles/examples.html">Examples</a>
    <a class="dropdown-item" href="../articles/inference_models.html">Inference Models</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/beautier/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="alpha_parameter_to_xml.html">alpha_parameter_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="are_clock_models.html">are_clock_models()</a></code> 
        </dt>
        <dd>Determine if x consists out of clock_models objects</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_mcmcs.html">are_equal_mcmcs()</a></code> 
        </dt>
        <dd>Determine if two MCMCs are equal.</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_screenlogs.html">are_equal_screenlogs()</a></code> 
        </dt>
        <dd>Determine if two screenlogs are equal.</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_tracelogs.html">are_equal_tracelogs()</a></code> 
        </dt>
        <dd>Determine if two tracelogs are equal.</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_treelogs.html">are_equal_treelogs()</a></code> 
        </dt>
        <dd>Determine if two treelogs are equal.</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_xml_files.html">are_equal_xml_files()</a></code> 
        </dt>
        <dd>Determine if XML files result in equal trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equal_xml_lines.html">are_equal_xml_lines()</a></code> 
        </dt>
        <dd>Determine if XML lines result in equal trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_files.html">are_equivalent_xml_files()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_lines.html">are_equivalent_xml_lines()</a></code> 
        </dt>
        <dd>Determine if XML lines result in equivalent trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_lines_all.html">are_equivalent_xml_lines_all()</a></code> 
        </dt>
        <dd>Determine if XML lines result in equivalent trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_lines_loggers.html">are_equivalent_xml_lines_loggers()</a></code> 
        </dt>
        <dd>Determine if XML operator lines result in equivalent trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_lines_operators.html">are_equivalent_xml_lines_operators()</a></code> 
        </dt>
        <dd>Determine if XML operator lines result in equivalent trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_equivalent_xml_lines_section.html">are_equivalent_xml_lines_section()</a></code> 
        </dt>
        <dd>Determine if XML lines result in equivalent trees</dd>
      </dl><dl><dt>
          
          <code><a href="are_fasta_filenames.html">are_fasta_filenames()</a></code> 
        </dt>
        <dd>Checks if all filenames have a FASTA filename extension</dd>
      </dl><dl><dt>
          
          <code><a href="are_ids.html">are_ids()</a></code> 
        </dt>
        <dd>Determine if x consists out of IDs</dd>
      </dl><dl><dt>
          
          <code><a href="are_init_clock_models.html">are_init_clock_models()</a></code> 
        </dt>
        <dd>Determine if x consists out of initialized clock_models objects</dd>
      </dl><dl><dt>
          
          <code><a href="are_init_mrca_priors.html">are_init_mrca_priors()</a></code> 
        </dt>
        <dd>Determine if x consists out of initialized MRCA priors</dd>
      </dl><dl><dt>
          
          <code><a href="are_init_site_models.html">are_init_site_models()</a></code> 
        </dt>
        <dd>Determine if x consists out of initialized site_models objects</dd>
      </dl><dl><dt>
          
          <code><a href="are_init_tree_priors.html">are_init_tree_priors()</a></code> 
        </dt>
        <dd>Determine if x consists out of initialized tree_priors objects</dd>
      </dl><dl><dt>
          
          <code><a href="are_mrca_align_ids_in_fasta.html">are_mrca_align_ids_in_fasta()</a></code> 
        </dt>
        <dd>Determine if the MRCA priors' alignment IDs are present in the FASTA files</dd>
      </dl><dl><dt>
          
          <code><a href="are_mrca_priors.html">are_mrca_priors()</a></code> 
        </dt>
        <dd>Determine if x consists out of MRCA priors</dd>
      </dl><dl><dt>
          
          <code><a href="are_mrca_taxon_names_in_fasta.html">are_mrca_taxon_names_in_fasta()</a></code> 
        </dt>
        <dd>Determine if the MRCA priors' taxa names are present in the FASTA files</dd>
      </dl><dl><dt>
          
          <code><a href="are_rln_clock_models.html">are_rln_clock_models()</a></code> 
        </dt>
        <dd>Are the clock models Relaxed Log-Normal clock models?</dd>
      </dl><dl><dt>
          
          <code><a href="are_site_models.html">are_site_models()</a></code> 
        </dt>
        <dd>Determine if x consists out of site_models objects</dd>
      </dl><dl><dt>
          
          <code><a href="are_tree_priors.html">are_tree_priors()</a></code> 
        </dt>
        <dd>Determine if x consists out of tree_priors objects</dd>
      </dl><dl><dt>
          
          <code><a href="b_pop_sizes_param_to_xml.html">b_pop_sizes_param_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="b_pop_sizes_parameter_to_xml.html">b_pop_sizes_parameter_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="bd_tree_prior_to_xml_prior_distr.html">bd_tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="beta_parameter_to_xml.html">beta_parameter_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="cbs_tree_prior_to_xml_prior_distr.html">cbs_tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Birth-Death tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="ccp_tree_prior_to_xml_prior_distr.html">ccp_tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Constant Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="ccp_tree_prior_to_xml_state.html">ccp_tree_prior_to_xml_state()</a></code> 
        </dt>
        <dd>Convert a CCP tree prior
to the XML as part of the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="cep_tree_prior_to_xml_prior_distr.html">cep_tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the tree prior section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a
Coalescent Exponential Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="check_alignment_id.html">check_alignment_id()</a></code> 
        </dt>
        <dd>Check if the <code>alignment_id</code> is valid.</dd>
      </dl><dl><dt>
          
          <code><a href="check_beauti_options.html">check_beauti_options()</a></code> 
        </dt>
        <dd>Check if the <code>beauti_options</code> is a valid <code>beauti_options</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_clock_model.html">check_clock_model()</a></code> 
        </dt>
        <dd>Check if the clock model is a valid clock model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_clock_models.html">check_clock_models()</a></code> 
        </dt>
        <dd>Check if the object is a list of one or more clock models.</dd>
      </dl><dl><dt>
          
          <code><a href="check_empty_beautier_folder.html">check_empty_beautier_folder()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="check_file_and_model_agree.html">check_file_and_model_agree()</a></code> 
        </dt>
        <dd>Checks if the input FASTA file and the inference model agree.</dd>
      </dl><dl><dt>
          
          <code><a href="check_file_exists.html">check_file_exists()</a></code> 
        </dt>
        <dd>Function to check if a file exists.
Calls <code>stop</code> if the file is absent</dd>
      </dl><dl><dt>
          
          <code><a href="check_filename.html">check_filename()</a></code> 
        </dt>
        <dd>Check if the `filename` is valid</dd>
      </dl><dl><dt>
          
          <code><a href="check_gamma_site_model.html">check_gamma_site_model()</a></code> 
        </dt>
        <dd>Checks if the parameter is a valid gamma site model</dd>
      </dl><dl><dt>
          
          <code><a href="check_gamma_site_model_names.html">check_gamma_site_model_names()</a></code> 
        </dt>
        <dd>Checks if the gamma site model has the right list elements' names</dd>
      </dl><dl><dt>
          
          <code><a href="check_gtr_site_model.html">check_gtr_site_model()</a></code> 
        </dt>
        <dd>Check if the <code>gtr_site_model</code> is a valid
GTR nucleotide substitution model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_gtr_site_model_names.html">check_gtr_site_model_names()</a></code> 
        </dt>
        <dd>Check if the <code>gtr_site_model</code> has the list elements
of a valid <code>gtr_site_model</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_inference_model.html">check_inference_model()</a></code> 
        </dt>
        <dd>Check if the supplied object is a valid
Bayesian phylogenetic inference model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_inference_models.html">check_inference_models()</a></code> 
        </dt>
        <dd>Check if the <code>inference_model</code> is a valid BEAUti inference model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_is_monophyletic.html">check_is_monophyletic()</a></code> 
        </dt>
        <dd>Check if <code>is_monophyletic</code> has a valid value.</dd>
      </dl><dl><dt>
          
          <code><a href="check_log_mode.html">check_log_mode()</a></code> 
        </dt>
        <dd>Check if the supplied <code>mode</code> is a valid logging mode.</dd>
      </dl><dl><dt>
          
          <code><a href="check_log_sort.html">check_log_sort()</a></code> 
        </dt>
        <dd>Check if the supplied <code>sort</code> is a valid logging sorting option.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mcmc.html">check_mcmc()</a></code> 
        </dt>
        <dd>Check if the MCMC is a valid MCMC object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mcmc_list_element_names.html">check_mcmc_list_element_names()</a></code> 
        </dt>
        <dd>Check if the MCMC has the list elements of a valid MCMC object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mcmc_values.html">check_mcmc_values()</a></code> 
        </dt>
        <dd>Check if the MCMC has the list elements with valid values
for being a valid MCMC object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mrca_prior.html">check_mrca_prior()</a></code> 
        </dt>
        <dd>Check if the MRCA prior is a valid MRCA prior.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mrca_prior_name.html">check_mrca_prior_name()</a></code> 
        </dt>
        <dd>Check if <code>mrca_prior_name</code> is a valid MRCA prior name.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mrca_prior_names.html">check_mrca_prior_names()</a></code> 
        </dt>
        <dd>Check if the MRCA prior,
which is a list, has all the named elements.</dd>
      </dl><dl><dt>
          
          <code><a href="check_mrca_prior_taxa_names.html">check_mrca_prior_taxa_names()</a></code> 
        </dt>
        <dd>Check the MRCA prior's taxon names are valid.</dd>
      </dl><dl><dt>
          
          <code><a href="check_ns_mcmc.html">check_ns_mcmc()</a></code> 
        </dt>
        <dd>Check if this an MCMC that uses Nested Sampling
to estimate a marginal likelihood.</dd>
      </dl><dl><dt>
          
          <code><a href="check_param.html">check_param()</a></code> 
        </dt>
        <dd>Check if the parameter is a valid parameter</dd>
      </dl><dl><dt>
          
          <code><a href="check_param_names.html">check_param_names()</a></code> 
        </dt>
        <dd>Check if the <code>param</code> has the list elements
of a valid <code>param</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_param_types.html">check_param_types()</a></code> 
        </dt>
        <dd>Check if the <code>param</code> has the list elements
of the right type for a valid <code>param</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_phylogeny.html">check_phylogeny()</a></code> 
        </dt>
        <dd>Check if the phylogeny is a valid phylogeny object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_rename_fun.html">check_rename_fun()</a></code> 
        </dt>
        <dd>Check if the rename function is a valid filename rename function</dd>
      </dl><dl><dt>
          
          <code><a href="check_rln_clock_model.html">check_rln_clock_model()</a></code> 
        </dt>
        <dd>Check if the clock model is a valid clock model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_screenlog.html">check_screenlog()</a></code> 
        </dt>
        <dd>Check if a <code>screenlog</code> is valid.</dd>
      </dl><dl><dt>
          
          <code><a href="check_screenlog_names.html">check_screenlog_names()</a></code> 
        </dt>
        <dd>Check if the <code>screenlog</code> has the list elements
of a valid <code>screenlog</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_screenlog_values.html">check_screenlog_values()</a></code> 
        </dt>
        <dd>Check if the screenlog has the list elements with valid values
for being a valid screenlog object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_site_model.html">check_site_model()</a></code> 
        </dt>
        <dd>Check if the site model is a valid site model</dd>
      </dl><dl><dt>
          
          <code><a href="check_site_model_names.html">check_site_model_names()</a></code> 
        </dt>
        <dd>Check if the <code>site_model</code> has the list elements
of a valid <code>site_model</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_site_model_types.html">check_site_model_types()</a></code> 
        </dt>
        <dd>Check if the <code>site_model</code> has the list elements
of the right type for a valid <code>site_model</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_site_models.html">check_site_models()</a></code> 
        </dt>
        <dd>Check if the object is a list of one or more site models.</dd>
      </dl><dl><dt>
          
          <code><a href="check_store_every.html">check_store_every()</a></code> 
        </dt>
        <dd>Check if <code>store_every</code> holds a valid value</dd>
      </dl><dl><dt>
          
          <code><a href="check_strict_clock_model.html">check_strict_clock_model()</a></code> 
        </dt>
        <dd>Check if the clock model is a valid clock model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tn93_site_model.html">check_tn93_site_model()</a></code> 
        </dt>
        <dd>Check if the <code>tn93_site_model</code> is a valid
TN93 nucleotide substitution model.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tn93_site_model_names.html">check_tn93_site_model_names()</a></code> 
        </dt>
        <dd>Check if the <code>tn93_site_model</code> has the list elements
of a valid <code>tn93_site_model</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tracelog.html">check_tracelog()</a></code> 
        </dt>
        <dd>Check if a <code>tracelog</code> is valid.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tracelog_names.html">check_tracelog_names()</a></code> 
        </dt>
        <dd>Check if the <code>tracelog</code> has the list elements
of a valid <code>tracelog</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tracelog_values.html">check_tracelog_values()</a></code> 
        </dt>
        <dd>Check if the tracelog has the list elements with valid values
for being a valid tracelog object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_tree_prior.html">check_tree_prior()</a></code> 
        </dt>
        <dd>Check if the tree prior is a valid tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="check_tree_priors.html">check_tree_priors()</a></code> 
        </dt>
        <dd>Check if the object is a list of one or more tree priors.</dd>
      </dl><dl><dt>
          
          <code><a href="check_treelog.html">check_treelog()</a></code> 
        </dt>
        <dd>Check if a <code>treelog</code> is valid.</dd>
      </dl><dl><dt>
          
          <code><a href="check_treelog_names.html">check_treelog_names()</a></code> 
        </dt>
        <dd>Check if the <code>treelog</code> has the list elements
of a valid <code>treelog</code> object.</dd>
      </dl><dl><dt>
          
          <code><a href="check_treelog_values.html">check_treelog_values()</a></code> 
        </dt>
        <dd>Check if the treelog has the list elements with valid values
for being a valid treelog object.</dd>
      </dl><dl><dt>
          
          <code><a href="clock_model_to_xml_operators.html">clock_model_to_xml_operators()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="clock_model_to_xml_prior_distr.html">clock_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="clock_model_to_xml_state.html">clock_model_to_xml_state()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="clock_model_to_xml_tracelog.html">clock_model_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="clock_model_to_xml_treelogger.html">clock_model_to_xml_treelogger()</a></code> 
        </dt>
        <dd>Convert a clock model to the XML of the <code>TreeLogger</code></dd>
      </dl><dl><dt>
          
          <code><a href="clock_rate_param_to_xml.html">clock_rate_param_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="compare_lines.html">compare_lines()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="count_trailing_spaces.html">count_trailing_spaces()</a></code> 
        </dt>
        <dd>Count the number of spaces before the first character</dd>
      </dl><dl><dt>
          
          <code><a href="create_alpha_param.html">create_alpha_param()</a></code> 
        </dt>
        <dd>Create a parameter called alpha</dd>
      </dl><dl><dt>
          
          <code><a href="create_b_pop_sizes_param.html">create_b_pop_sizes_param()</a></code> 
        </dt>
        <dd>Create a parameter called `b_pop_sizes`.</dd>
      </dl><dl><dt>
          
          <code><a href="create_bd_tree_prior.html">create_bd_tree_prior()</a></code> 
        </dt>
        <dd>Create a Birth-Death tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_beast_xml.html">create_beast2_beast_xml()</a></code> 
        </dt>
        <dd>Create the <code>&lt;beast ...&gt;</code> XML</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input.html">create_beast2_input()</a></code> 
        </dt>
        <dd>Create a BEAST2 XML input text</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_beast.html">create_beast2_input_beast()</a></code> 
        </dt>
        <dd>Creates the XML text for the <code>beast</code> tag of a BEAST2 parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_data.html">create_beast2_input_data()</a></code> 
        </dt>
        <dd>Creates the <code>data</code> section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_data_sequences.html">create_beast2_input_data_sequences()</a></code> 
        </dt>
        <dd>Creates the data section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_distr.html">create_beast2_input_distr()</a></code> 
        </dt>
        <dd>Creates the distribution section of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_distr_lh.html">create_beast2_input_distr_lh()</a></code> 
        </dt>
        <dd>Creates the XML text for the <code>distribution</code> tag
with the <code>likelihood</code> ID,
of a BEAST2 parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_distr_prior.html">create_beast2_input_distr_prior()</a></code> 
        </dt>
        <dd>Creates the prior section in the distribution section
of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_file.html">create_beast2_input_file()</a></code> 
        </dt>
        <dd>Create a BEAST2 input file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_file_from_model.html">create_beast2_input_file_from_model()</a></code> 
        </dt>
        <dd>Create a BEAST2 input file from an inference model</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_from_model.html">create_beast2_input_from_model()</a></code> 
        </dt>
        <dd>Create a BEAST2 XML input text from an inference model</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_init.html">create_beast2_input_init()</a></code> 
        </dt>
        <dd>Creates the <code>init</code> section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_map.html">create_beast2_input_map()</a></code> 
        </dt>
        <dd>Creates the map section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_operators.html">create_beast2_input_operators()</a></code> 
        </dt>
        <dd>Creates the operators section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_run.html">create_beast2_input_run()</a></code> 
        </dt>
        <dd>Creates the '<code>run</code>' section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beast2_input_state.html">create_beast2_input_state()</a></code> 
        </dt>
        <dd>Creates the '<code>state</code>' section of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_beauti_options.html">create_beauti_options()</a></code> 
        </dt>
        <dd>Function to create a set of `BEAUti` options.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beauti_options_v2_4.html">create_beauti_options_v2_4()</a></code> 
        </dt>
        <dd>Function to create the BEAUti options for version 2.4.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beauti_options_v2_6.html">create_beauti_options_v2_6()</a></code> 
        </dt>
        <dd>Function to create the BEAUti options for version 2.6.</dd>
      </dl><dl><dt>
          
          <code><a href="create_beautier_tempfolder.html">create_beautier_tempfolder()</a></code> 
        </dt>
        <dd>Create the default `beautier` temporary folder</dd>
      </dl><dl><dt>
          
          <code><a href="create_beta_distr.html">create_beta_distr()</a></code> 
        </dt>
        <dd>Create a beta distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_beta_param.html">create_beta_param()</a></code> 
        </dt>
        <dd>Create a parameter called beta</dd>
      </dl><dl><dt>
          
          <code><a href="create_branch_rate_model_xml.html">create_branch_rate_model_xml()</a></code> 
        </dt>
        <dd>Internal function to create the <code>branchRateModel</code> section
of the XML as text.</dd>
      </dl><dl><dt>
          
          <code><a href="create_cbs_tree_prior.html">create_cbs_tree_prior()</a></code> 
        </dt>
        <dd>Create a Coalescent Bayesian Skyline tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_ccp_tree_prior.html">create_ccp_tree_prior()</a></code> 
        </dt>
        <dd>Create a Coalescent Constant Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_cep_tree_prior.html">create_cep_tree_prior()</a></code> 
        </dt>
        <dd>Create a Coalescent Exponential Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_model.html">create_clock_model()</a></code> 
        </dt>
        <dd>General function to create a clock model</dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_model_from_name.html">create_clock_model_from_name()</a></code> 
        </dt>
        <dd>Create a clock model from name</dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_models.html">create_clock_models()</a></code> 
        </dt>
        <dd>Creates all supported clock models,
  which is a list of the types returned by
  <code>create_rln_clock_model</code>,
  and <code>create_strict_clock_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_models_from_names.html">create_clock_models_from_names()</a></code> 
        </dt>
        <dd>Create clock models from their names</dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_rate_param.html">create_clock_rate_param()</a></code> 
        </dt>
        <dd>Create a parameter called <code>clock_rate</code>,
  as needed by <code>create_strict_clock_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="create_clock_rate_state_node_parameter_xml.html">create_clock_rate_state_node_parameter_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_data_xml.html">create_data_xml()</a></code> 
        </dt>
        <dd>Create the <code>&lt;data ..&gt;</code> XML</dd>
      </dl><dl><dt>
          
          <code><a href="create_distr.html">create_distr()</a></code> 
        </dt>
        <dd>General function to create a distribution.</dd>
      </dl><dl><dt>
          
          <code><a href="create_exp_distr.html">create_exp_distr()</a></code> 
        </dt>
        <dd>Create an exponential distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_freq_param.html">create_freq_param()</a></code> 
        </dt>
        <dd>Create a parameter called freq</dd>
      </dl><dl><dt>
          
          <code><a href="create_gamma_distr.html">create_gamma_distr()</a></code> 
        </dt>
        <dd>Create a gamma distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_gamma_site_model.html">create_gamma_site_model()</a></code> 
        </dt>
        <dd>Create a gamma site model, part of a site model</dd>
      </dl><dl><dt>
          
          <code><a href="create_gtr_site_model.html">create_gtr_site_model()</a></code> 
        </dt>
        <dd>Create a GTR site model</dd>
      </dl><dl><dt>
          
          <code><a href="create_gtr_subst_model_xml.html">create_gtr_subst_model_xml()</a></code> 
        </dt>
        <dd>Converts a GTR site model to XML,
  used in the <code>substModel</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="create_hky_site_model.html">create_hky_site_model()</a></code> 
        </dt>
        <dd>Create an HKY site model</dd>
      </dl><dl><dt>
          
          <code><a href="create_hky_subst_model_xml.html">create_hky_subst_model_xml()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>substModel</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="create_inference_model.html">create_inference_model()</a></code> 
        </dt>
        <dd>Create a Bayesian phylogenetic inference model.</dd>
      </dl><dl><dt>
          
          <code><a href="create_inv_gamma_distr.html">create_inv_gamma_distr()</a></code> 
        </dt>
        <dd>Create an inverse-gamma distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_jc69_site_model.html">create_jc69_site_model()</a></code> 
        </dt>
        <dd>Create a JC69 site model</dd>
      </dl><dl><dt>
          
          <code><a href="create_jc69_subst_model_xml.html">create_jc69_subst_model_xml()</a></code> 
        </dt>
        <dd>Converts a JC69 site model to XML,
used in the <code>substModel</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="create_kappa_1_param.html">create_kappa_1_param()</a></code> 
        </dt>
        <dd>Create a parameter called kappa 1</dd>
      </dl><dl><dt>
          
          <code><a href="create_kappa_2_param.html">create_kappa_2_param()</a></code> 
        </dt>
        <dd>Create a parameter called kappa 2</dd>
      </dl><dl><dt>
          
          <code><a href="create_kappa_param.html">create_kappa_param()</a></code> 
        </dt>
        <dd>Create a parameter called kappa</dd>
      </dl><dl><dt>
          
          <code><a href="create_lambda_param.html">create_lambda_param()</a></code> 
        </dt>
        <dd>Create a parameter called lambda</dd>
      </dl><dl><dt>
          
          <code><a href="create_laplace_distr.html">create_laplace_distr()</a></code> 
        </dt>
        <dd>Create a Laplace distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_log_normal_distr.html">create_log_normal_distr()</a></code> 
        </dt>
        <dd>Create a log-normal distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_loggers_xml.html">create_loggers_xml()</a></code> 
        </dt>
        <dd>Creates the three logger sections of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_m_param.html">create_m_param()</a></code> 
        </dt>
        <dd>Create a parameter called m</dd>
      </dl><dl><dt>
          
          <code><a href="create_mcmc.html">create_mcmc()</a></code> 
        </dt>
        <dd>Create an MCMC configuration.</dd>
      </dl><dl><dt>
          
          <code><a href="create_mean_param.html">create_mean_param()</a></code> 
        </dt>
        <dd>Create a parameter called mean</dd>
      </dl><dl><dt>
          
          <code><a href="create_mrca_prior.html">create_mrca_prior()</a></code> 
        </dt>
        <dd>Create a Most Recent Common Ancestor prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_mu_param.html">create_mu_param()</a></code> 
        </dt>
        <dd>Create a parameter called mu</dd>
      </dl><dl><dt>
          
          <code><a href="create_normal_distr.html">create_normal_distr()</a></code> 
        </dt>
        <dd>Create an normal distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_ns_inference_model.html">create_ns_inference_model()</a></code> 
        </dt>
        <dd>Create an inference model to measure the evidence of.</dd>
      </dl><dl><dt>
          
          <code><a href="create_ns_mcmc.html">create_ns_mcmc()</a></code> 
        </dt>
        <dd>Create an MCMC object to estimate the marginal likelihood
using Nested Sampling.</dd>
      </dl><dl><dt>
          
          <code><a href="create_one_div_x_distr.html">create_one_div_x_distr()</a></code> 
        </dt>
        <dd>Create a 1/x distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_param.html">create_param()</a></code> 
        </dt>
        <dd>General function to create a parameter.</dd>
      </dl><dl><dt>
          
          <code><a href="create_poisson_distr.html">create_poisson_distr()</a></code> 
        </dt>
        <dd>Create a Poisson distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_ac_param.html">create_rate_ac_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate AC'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_ag_param.html">create_rate_ag_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate AG'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_at_param.html">create_rate_at_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate AT'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_categories_state_node_xml.html">create_rate_categories_state_node_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_cg_param.html">create_rate_cg_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate CG'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_ct_param.html">create_rate_ct_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate CT'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rate_gt_param.html">create_rate_gt_param()</a></code> 
        </dt>
        <dd>Create a parameter called 'rate GT'</dd>
      </dl><dl><dt>
          
          <code><a href="create_rln_clock_branch_rate_model_xml.html">create_rln_clock_branch_rate_model_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_rln_clock_model.html">create_rln_clock_model()</a></code> 
        </dt>
        <dd>Create a relaxed log-normal clock model</dd>
      </dl><dl><dt>
          
          <code><a href="create_s_param.html">create_s_param()</a></code> 
        </dt>
        <dd>Create a parameter called s</dd>
      </dl><dl><dt>
          
          <code><a href="create_scale_param.html">create_scale_param()</a></code> 
        </dt>
        <dd>Create a parameter called scale</dd>
      </dl><dl><dt>
          
          <code><a href="create_screenlog.html">create_screenlog()</a></code> 
        </dt>
        <dd>Create a <code>screenlog</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="create_screenlog_xml.html">create_screenlog_xml()</a></code> 
        </dt>
        <dd>Creates the <code>screenlog</code> section of the <code>logger</code> section
of a BEAST2 XML parameter file</dd>
      </dl><dl><dt>
          
          <code><a href="create_sigma_param.html">create_sigma_param()</a></code> 
        </dt>
        <dd>Create a parameter called sigma</dd>
      </dl><dl><dt>
          
          <code><a href="create_site_model.html">create_site_model()</a></code> 
        </dt>
        <dd>General function to create a site model.</dd>
      </dl><dl><dt>
          
          <code><a href="create_site_model_from_name.html">create_site_model_from_name()</a></code> 
        </dt>
        <dd>Create a site model from name</dd>
      </dl><dl><dt>
          
          <code><a href="create_site_model_parameters_xml.html">create_site_model_parameters_xml()</a></code> 
        </dt>
        <dd>Internal function to creates the XML text for the
<code>parameter</code>s within the <code>siteModel</code> section
of a BEAST2 parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_site_model_xml.html">create_site_model_xml()</a></code> 
        </dt>
        <dd>Internal function to creates the XML text for the <code>siteModel</code> tag
of a BEAST2 parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_site_models.html">create_site_models()</a></code> 
        </dt>
        <dd>Creates all supported site models
  which is a list of the types returned by
  <code>create_gtr_site_model</code>,
  <code>create_hky_site_model</code>,
  <code>create_jc69_site_model</code>
  and <code>create_tn93_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="create_site_models_from_names.html">create_site_models_from_names()</a></code> 
        </dt>
        <dd>Create site models from their names</dd>
      </dl><dl><dt>
          
          <code><a href="create_strict_clock_branch_rate_model_xml.html">create_strict_clock_branch_rate_model_xml()</a></code> 
        </dt>
        <dd>Internal function.</dd>
      </dl><dl><dt>
          
          <code><a href="create_strict_clock_model.html">create_strict_clock_model()</a></code> 
        </dt>
        <dd>Create a strict clock model</dd>
      </dl><dl><dt>
          
          <code><a href="create_strict_clock_rate_scaler_operator_xml.html">create_strict_clock_rate_scaler_operator_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_subst_model_xml.html">create_subst_model_xml()</a></code> 
        </dt>
        <dd>Internal function to create the <code>substModel</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="create_temp_screenlog_filename.html">create_temp_screenlog_filename()</a></code> 
        </dt>
        <dd>Create a filename for a temporary `screenlog` file</dd>
      </dl><dl><dt>
          
          <code><a href="create_temp_tracelog_filename.html">create_temp_tracelog_filename()</a></code> 
        </dt>
        <dd>Create a filename for a temporary `tracelog` file</dd>
      </dl><dl><dt>
          
          <code><a href="create_temp_treelog_filename.html">create_temp_treelog_filename()</a></code> 
        </dt>
        <dd>Create a filename for a temporary `treelog` file</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_inference_model.html">create_test_inference_model()</a></code> 
        </dt>
        <dd>Create a testing inference model.</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_mcmc.html">create_test_mcmc()</a></code> 
        </dt>
        <dd>Create an MCMC configuration for testing.</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_ns_inference_model.html">create_test_ns_inference_model()</a></code> 
        </dt>
        <dd>Create an inference model to be tested by Nested Sampling</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_ns_mcmc.html">create_test_ns_mcmc()</a></code> 
        </dt>
        <dd>Create an NS MCMC object for testing</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_screenlog.html">create_test_screenlog()</a></code> 
        </dt>
        <dd>Create a <code>screenlog</code> object, to be used in testing</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_tracelog.html">create_test_tracelog()</a></code> 
        </dt>
        <dd>Create a <code>tracelog</code> object, as used for testing</dd>
      </dl><dl><dt>
          
          <code><a href="create_test_treelog.html">create_test_treelog()</a></code> 
        </dt>
        <dd>Create a <code>treelog</code> object to be used in testing</dd>
      </dl><dl><dt>
          
          <code><a href="create_tn93_site_model.html">create_tn93_site_model()</a></code> 
        </dt>
        <dd>Create a TN93 site model</dd>
      </dl><dl><dt>
          
          <code><a href="create_tn93_subst_model_xml.html">create_tn93_subst_model_xml()</a></code> 
        </dt>
        <dd>Converts a TN93 site model to XML,
  used in the <code>substModel</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="create_tracelog.html">create_tracelog()</a></code> 
        </dt>
        <dd>Create a <code>tracelog</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="create_tracelog_xml.html">create_tracelog_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_trait_set_string.html">create_trait_set_string()</a></code> 
        </dt>
        <dd>Create a trait set string.</dd>
      </dl><dl><dt>
          
          <code><a href="create_tree_likelihood_distr_xml.html">create_tree_likelihood_distr_xml()</a></code> 
        </dt>
        <dd>Creates the XML text for the <code>distribution</code> tag
with the <code>treeLikelihood</code> ID,
of a BEAST2 parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="create_tree_prior.html">create_tree_prior()</a></code> 
        </dt>
        <dd>Internal function to create a tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="create_tree_priors.html">create_tree_priors()</a></code> 
        </dt>
        <dd>Creates all supported tree priors,
  which is a list of the types returned by
  <code>create_bd_tree_prior</code>,
  <code>create_cbs_tree_prior</code>,
  <code>create_ccp_tree_prior</code>,
  <code>create_cep_tree_prior</code>
  and <code>create_yule_tree_prior</code></dd>
      </dl><dl><dt>
          
          <code><a href="create_treelog.html">create_treelog()</a></code> 
        </dt>
        <dd>Create a <code>treelog</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="create_treelog_xml.html">create_treelog_xml()</a></code> 
        </dt>
        <dd>Creates the XML text for the `logger` tag with ID `treelog`.
This section has these elements:
```
&lt;logger id="treelog.t:test_output_0" spec="Logger" fileName="my_treelog.trees" logEvery="345000" mode="tree" sanitiseHeaders="true" sort="smart"&gt; # nolint indeed long
    &lt;log id="TreeWithMetaDataLogger.t:test_output_0" spec="beast.evolution.tree.TreeWithMetaDataLogger" tree="@Tree.t:test_output_0"/&gt; # nolint indeed long
&lt;/logger&gt;
```</dd>
      </dl><dl><dt>
          
          <code><a href="create_ucld_mean_state_node_param_xml.html">create_ucld_mean_state_node_param_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_ucld_stdev_state_node_param_xml.html">create_ucld_stdev_state_node_param_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="create_uniform_distr.html">create_uniform_distr()</a></code> 
        </dt>
        <dd>Create a uniform distribution</dd>
      </dl><dl><dt>
          
          <code><a href="create_xml_declaration.html">create_xml_declaration()</a></code> 
        </dt>
        <dd>Create the XML declaration of the BEAST2 XML input file</dd>
      </dl><dl><dt>
          
          <code><a href="create_yule_tree_prior.html">create_yule_tree_prior()</a></code> 
        </dt>
        <dd>Create a Yule tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="default_parameters_doc.html">default_parameters_doc()</a></code> 
        </dt>
        <dd>Documentation of parameters (for example, <code>create_param</code>.
This function does nothing. It is intended to inherit documentation from.</dd>
      </dl><dl><dt>
          
          <code><a href="default_params_doc.html">default_params_doc()</a></code> 
        </dt>
        <dd>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml.html">distr_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_beta.html">distr_to_xml_beta()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_exp.html">distr_to_xml_exp()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_inv_gamma.html">distr_to_xml_inv_gamma()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_laplace.html">distr_to_xml_laplace()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_log_normal.html">distr_to_xml_log_normal()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_normal.html">distr_to_xml_normal()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_one_div_x.html">distr_to_xml_one_div_x()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_poisson.html">distr_to_xml_poisson()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="distr_to_xml_uniform.html">distr_to_xml_uniform()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="extract_xml_loggers_from_lines.html">extract_xml_loggers_from_lines()</a></code> 
        </dt>
        <dd>Extract everything between first loggers and last loggers line</dd>
      </dl><dl><dt>
          
          <code><a href="extract_xml_operators_from_lines.html">extract_xml_operators_from_lines()</a></code> 
        </dt>
        <dd>Extract everything between first operators and last operators line</dd>
      </dl><dl><dt>
          
          <code><a href="extract_xml_section_from_lines.html">extract_xml_section_from_lines()</a></code> 
        </dt>
        <dd>Get the lines of an XML section, including the section tags</dd>
      </dl><dl><dt>
          
          <code><a href="fasta_file_to_sequences.html">fasta_file_to_sequences()</a></code> 
        </dt>
        <dd>Convert a FASTA file to a table of sequences</dd>
      </dl><dl><dt>
          
          <code><a href="find_clock_model.html">find_clock_model()</a></code> 
        </dt>
        <dd>Finds a clock model with a certain ID</dd>
      </dl><dl><dt>
          
          <code><a href="find_first_regex_line.html">find_first_regex_line()</a></code> 
        </dt>
        <dd>Find the first line that satisfies a regex</dd>
      </dl><dl><dt>
          
          <code><a href="find_first_xml_opening_tag_line.html">find_first_xml_opening_tag_line()</a></code> 
        </dt>
        <dd>Find the line number of the first section's opening tag</dd>
      </dl><dl><dt>
          
          <code><a href="find_last_regex_line.html">find_last_regex_line()</a></code> 
        </dt>
        <dd>Find the index of the last line that matches a regex</dd>
      </dl><dl><dt>
          
          <code><a href="find_last_xml_closing_tag_line.html">find_last_xml_closing_tag_line()</a></code> 
        </dt>
        <dd>Find the highest line number of a section's closing tag</dd>
      </dl><dl><dt>
          
          <code><a href="freq_equilibrium_to_xml.html">freq_equilibrium_to_xml()</a></code> 
        </dt>
        <dd>Creates the <code>freq_equilibrium</code> as XML</dd>
      </dl><dl><dt>
          
          <code><a href="freq_param_to_xml.html">freq_param_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="gamma_distr_to_xml.html">gamma_distr_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="gamma_site_model_to_xml_prior_distr.html">gamma_site_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function.</dd>
      </dl><dl><dt>
          
          <code><a href="gamma_site_model_to_xml_state.html">gamma_site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Converts a gamma site model to XML,
  used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="gamma_site_models_to_xml_prior_distr.html">gamma_site_models_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Deprecated function</dd>
      </dl><dl><dt>
          
          <code><a href="get_alignment_id.html">get_alignment_id()</a></code> 
        </dt>
        <dd>Conclude the ID from a FASTA filename.</dd>
      </dl><dl><dt>
          
          <code><a href="get_alignment_ids.html">get_alignment_ids()</a></code> 
        </dt>
        <dd>Get the alignment IDs from one or more files.</dd>
      </dl><dl><dt>
          
          <code><a href="get_alignment_ids_from_fasta_filenames.html">get_alignment_ids_from_fasta_filenames()</a></code> 
        </dt>
        <dd>Get the alignment ID from one or more FASTA filenames.</dd>
      </dl><dl><dt>
          
          <code><a href="get_beautier_folder.html">get_beautier_folder()</a></code> 
        </dt>
        <dd>Get the path to the beautier temporary files folder</dd>
      </dl><dl><dt>
          
          <code><a href="get_beautier_path.html">get_beautier_path()</a></code> 
        </dt>
        <dd>Get the full path of a file in the <code>inst/extdata</code> folder</dd>
      </dl><dl><dt>
          
          <code><a href="get_beautier_paths.html">get_beautier_paths()</a></code> 
        </dt>
        <dd>Get the full paths of files in the <code>inst/extdata</code> folder</dd>
      </dl><dl><dt>
          
          <code><a href="get_beautier_tempfilename.html">get_beautier_tempfilename()</a></code> 
        </dt>
        <dd>Get a temporary filename</dd>
      </dl><dl><dt>
          
          <code><a href="get_clock_model_name.html">get_clock_model_name()</a></code> 
        </dt>
        <dd>Get the BEAUti name for a clock model</dd>
      </dl><dl><dt>
          
          <code><a href="get_clock_model_names.html">get_clock_model_names()</a></code> 
        </dt>
        <dd>Get the clock model names</dd>
      </dl><dl><dt>
          
          <code><a href="get_clock_models_ids.html">get_clock_models_ids()</a></code> 
        </dt>
        <dd>Collect the IDs of the list of clock models</dd>
      </dl><dl><dt>
          
          <code><a href="get_crown_age.html">get_crown_age()</a></code> 
        </dt>
        <dd>Obtain the crown age of a phylogeny.</dd>
      </dl><dl><dt>
          
          <code><a href="get_default_beast_namespace.html">get_default_beast_namespace()</a></code> 
        </dt>
        <dd>Get the default `namespace` element value of the `beast` XML tag.</dd>
      </dl><dl><dt>
          
          <code><a href="get_default_beast_namespace_v2_4.html">get_default_beast_namespace_v2_4()</a></code> 
        </dt>
        <dd>Get the default `namespace` element value of the `beast` XML tag
for BEAST 2.4</dd>
      </dl><dl><dt>
          
          <code><a href="get_default_beast_namespace_v2_6.html">get_default_beast_namespace_v2_6()</a></code> 
        </dt>
        <dd>Get the default `namespace` element value of the `beast` XML tag
for BEAST 2.6</dd>
      </dl><dl><dt>
          
          <code><a href="get_distr_n_params.html">get_distr_n_params()</a></code> 
        </dt>
        <dd>Get the number of parameters a distribution uses</dd>
      </dl><dl><dt>
          
          <code><a href="get_distr_names.html">get_distr_names()</a></code> 
        </dt>
        <dd>Get the distribution names</dd>
      </dl><dl><dt>
          
          <code><a href="get_fasta_filename.html">get_fasta_filename()</a></code> 
        </dt>
        <dd>Get the path of a FASTA file used in testing</dd>
      </dl><dl><dt>
          
          <code><a href="get_file_base_sans_ext.html">get_file_base_sans_ext()</a></code> 
        </dt>
        <dd>Get the base of the filename base without extension</dd>
      </dl><dl><dt>
          
          <code><a href="get_freq_equilibrium_names.html">get_freq_equilibrium_names()</a></code> 
        </dt>
        <dd>Returns valid values for the <code>freq_equilibrium</code> argument</dd>
      </dl><dl><dt>
          
          <code><a href="get_gamma_site_model_n_distrs.html">get_gamma_site_model_n_distrs()</a></code> 
        </dt>
        <dd>Get the number of distributions in a gamma site model</dd>
      </dl><dl><dt>
          
          <code><a href="get_gamma_site_model_n_params.html">get_gamma_site_model_n_params()</a></code> 
        </dt>
        <dd>Get the number of distributions a site model has</dd>
      </dl><dl><dt>
          
          <code><a href="get_has_non_strict_clock_model.html">get_has_non_strict_clock_model()</a></code> 
        </dt>
        <dd>Determines if there is at least one non-strict clock model
in the list of one or more clock models</dd>
      </dl><dl><dt>
          
          <code><a href="get_inference_model_filenames.html">get_inference_model_filenames()</a></code> 
        </dt>
        <dd>Get the filenames stored in an inference model.</dd>
      </dl><dl><dt>
          
          <code><a href="get_log_modes.html">get_log_modes()</a></code> 
        </dt>
        <dd>Get the possible log modes</dd>
      </dl><dl><dt>
          
          <code><a href="get_log_sorts.html">get_log_sorts()</a></code> 
        </dt>
        <dd>Get the possible log sorts</dd>
      </dl><dl><dt>
          
          <code><a href="get_mcmc_filenames.html">get_mcmc_filenames()</a></code> 
        </dt>
        <dd>Get the filenames stored in an MCMC.</dd>
      </dl><dl><dt>
          
          <code><a href="get_n_taxa.html">get_n_taxa()</a></code> 
        </dt>
        <dd>Extract the number of taxa from a file</dd>
      </dl><dl><dt>
          
          <code><a href="get_operator_id_pre.html">get_operator_id_pre()</a></code> 
        </dt>
        <dd>Get the prefix of operator IDs</dd>
      </dl><dl><dt>
          
          <code><a href="get_param_names.html">get_param_names()</a></code> 
        </dt>
        <dd>Get the parameter names</dd>
      </dl><dl><dt>
          
          <code><a href="get_remove_dir_fun.html">get_remove_dir_fun()</a></code> 
        </dt>
        <dd>Get a function that, from a filename, returns the part
without the directory.</dd>
      </dl><dl><dt>
          
          <code><a href="get_remove_hex_fun.html">get_remove_hex_fun()</a></code> 
        </dt>
        <dd>Get a function that removes the hex string from filenames.</dd>
      </dl><dl><dt>
          
          <code><a href="get_replace_dir_fun.html">get_replace_dir_fun()</a></code> 
        </dt>
        <dd>Get a function to replace the directory of a filename</dd>
      </dl><dl><dt>
          
          <code><a href="get_site_model_n_distrs.html">get_site_model_n_distrs()</a></code> 
        </dt>
        <dd>Get the number of distributions a site model has</dd>
      </dl><dl><dt>
          
          <code><a href="get_site_model_n_params.html">get_site_model_n_params()</a></code> 
        </dt>
        <dd>Get the number of distributions a site model has</dd>
      </dl><dl><dt>
          
          <code><a href="get_site_model_names.html">get_site_model_names()</a></code> 
        </dt>
        <dd>Get the site models' names</dd>
      </dl><dl><dt>
          
          <code><a href="get_site_models_n_distrs.html">get_site_models_n_distrs()</a></code> 
        </dt>
        <dd>Get the number of distributions a site model has</dd>
      </dl><dl><dt>
          
          <code><a href="get_site_models_n_params.html">get_site_models_n_params()</a></code> 
        </dt>
        <dd>Get the number of distributions one or more site models have</dd>
      </dl><dl><dt>
          
          <code><a href="get_taxa_names.html">get_taxa_names()</a></code> 
        </dt>
        <dd>Extract the names of taxa from a file</dd>
      </dl><dl><dt>
          
          <code><a href="get_tree_prior_n_distrs.html">get_tree_prior_n_distrs()</a></code> 
        </dt>
        <dd>Get the number of distributions a tree prior has</dd>
      </dl><dl><dt>
          
          <code><a href="get_tree_prior_n_params.html">get_tree_prior_n_params()</a></code> 
        </dt>
        <dd>Get the number of parameters a tree prior has</dd>
      </dl><dl><dt>
          
          <code><a href="get_tree_prior_names.html">get_tree_prior_names()</a></code> 
        </dt>
        <dd>Get the tree prior names</dd>
      </dl><dl><dt>
          
          <code><a href="get_tree_priors_n_distrs.html">get_tree_priors_n_distrs()</a></code> 
        </dt>
        <dd>Get the number of distributions a tree prior has</dd>
      </dl><dl><dt>
          
          <code><a href="get_tree_priors_n_params.html">get_tree_priors_n_params()</a></code> 
        </dt>
        <dd>Get the number of parameters a list of tree priors has</dd>
      </dl><dl><dt>
          
          <code><a href="get_xml_closing_tag.html">get_xml_closing_tag()</a></code> 
        </dt>
        <dd>Get the XML closing tag</dd>
      </dl><dl><dt>
          
          <code><a href="get_xml_opening_tag.html">get_xml_opening_tag()</a></code> 
        </dt>
        <dd>Get the XML opening tag</dd>
      </dl><dl><dt>
          
          <code><a href="gtr_site_model_to_xml_prior_distr.html">gtr_site_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="gtr_site_model_to_xml_state.html">gtr_site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="has_mrca_prior.html">has_mrca_prior()</a></code> 
        </dt>
        <dd>Determines if the inference model has an MRCA prior.</dd>
      </dl><dl><dt>
          
          <code><a href="has_mrca_prior_with_distr.html">has_mrca_prior_with_distr()</a></code> 
        </dt>
        <dd>See if the inference model has one MRCA prior with a distribution</dd>
      </dl><dl><dt>
          
          <code><a href="has_rln_clock_model.html">has_rln_clock_model()</a></code> 
        </dt>
        <dd>Determine if the <code>inference_model</code> uses
a relaxed log-normal clock model.</dd>
      </dl><dl><dt>
          
          <code><a href="has_strict_clock_model.html">has_strict_clock_model()</a></code> 
        </dt>
        <dd>Determine if the <code>inference_model</code> uses a strict clock model.</dd>
      </dl><dl><dt>
          
          <code><a href="has_tip_dating.html">has_tip_dating()</a></code> 
        </dt>
        <dd>Determine if the <code>inference_model</code> uses tip dating.</dd>
      </dl><dl><dt>
          
          <code><a href="has_xml_closing_tag.html">has_xml_closing_tag()</a></code> 
        </dt>
        <dd>Is an XML closing tag with the value of <code>section</code>
  present among the lines of
  the text?</dd>
      </dl><dl><dt>
          
          <code><a href="has_xml_opening_tag.html">has_xml_opening_tag()</a></code> 
        </dt>
        <dd>Is an XML opening tag with value 'section' present among the lines of
  the text?</dd>
      </dl><dl><dt>
          
          <code><a href="has_xml_short_closing_tag.html">has_xml_short_closing_tag()</a></code> 
        </dt>
        <dd>Is an XML closing tag with short closing text in
one of the lines of the text?</dd>
      </dl><dl><dt>
          
          <code><a href="hky_site_model_to_xml_prior_distr.html">hky_site_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="hky_site_model_to_xml_state.html">hky_site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="indent.html">indent()</a></code> 
        </dt>
        <dd>Indent text for a certain number of spaces.
If the text is only whitespace, leave it as such</dd>
      </dl><dl><dt>
          
          <code><a href="init_bd_tree_prior.html">init_bd_tree_prior()</a></code> 
        </dt>
        <dd>Initializes a Birth-Death tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="init_beta_distr.html">init_beta_distr()</a></code> 
        </dt>
        <dd>Initializes a beta distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_ccp_tree_prior.html">init_ccp_tree_prior()</a></code> 
        </dt>
        <dd>Initializes a Coalescent Constant Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="init_cep_tree_prior.html">init_cep_tree_prior()</a></code> 
        </dt>
        <dd>Initializes a Coalescent Exponential Population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="init_clock_models.html">init_clock_models()</a></code> 
        </dt>
        <dd>Initializes all clock models</dd>
      </dl><dl><dt>
          
          <code><a href="init_distr.html">init_distr()</a></code> 
        </dt>
        <dd>Initializes a distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_exp_distr.html">init_exp_distr()</a></code> 
        </dt>
        <dd>Initializes an exponential distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_gamma_distr.html">init_gamma_distr()</a></code> 
        </dt>
        <dd>Initializes a gamma distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_gamma_site_model.html">init_gamma_site_model()</a></code> 
        </dt>
        <dd>Initializes a gamma site model</dd>
      </dl><dl><dt>
          
          <code><a href="init_gtr_site_model.html">init_gtr_site_model()</a></code> 
        </dt>
        <dd>Initializes a GTR site model</dd>
      </dl><dl><dt>
          
          <code><a href="init_hky_site_model.html">init_hky_site_model()</a></code> 
        </dt>
        <dd>Initializes an HKY site model</dd>
      </dl><dl><dt>
          
          <code><a href="init_inference_model.html">init_inference_model()</a></code> 
        </dt>
        <dd>Initialize an inference model</dd>
      </dl><dl><dt>
          
          <code><a href="init_inv_gamma_distr.html">init_inv_gamma_distr()</a></code> 
        </dt>
        <dd>Initializes an inverse gamma distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_jc69_site_model.html">init_jc69_site_model()</a></code> 
        </dt>
        <dd>Initializes a JC69 site model</dd>
      </dl><dl><dt>
          
          <code><a href="init_laplace_distr.html">init_laplace_distr()</a></code> 
        </dt>
        <dd>Initializes an Laplace distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_log_normal_distr.html">init_log_normal_distr()</a></code> 
        </dt>
        <dd>Initializes an log-normal distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_mrca_prior.html">init_mrca_prior()</a></code> 
        </dt>
        <dd>Initialize the MRCA prior.</dd>
      </dl><dl><dt>
          
          <code><a href="init_mrca_priors.html">init_mrca_priors()</a></code> 
        </dt>
        <dd>Initializes all MRCA priors</dd>
      </dl><dl><dt>
          
          <code><a href="init_normal_distr.html">init_normal_distr()</a></code> 
        </dt>
        <dd>Initializes an normal distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_one_div_x_distr.html">init_one_div_x_distr()</a></code> 
        </dt>
        <dd>Initializes an one-divided-by-x distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_param.html">init_param()</a></code> 
        </dt>
        <dd>Initializes a parameter</dd>
      </dl><dl><dt>
          
          <code><a href="init_poisson_distr.html">init_poisson_distr()</a></code> 
        </dt>
        <dd>Initializes an Poisson distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_rln_clock_model.html">init_rln_clock_model()</a></code> 
        </dt>
        <dd>Initializes a Relaxed Log-Normal clock model</dd>
      </dl><dl><dt>
          
          <code><a href="init_site_models.html">init_site_models()</a></code> 
        </dt>
        <dd>Initializes all site models</dd>
      </dl><dl><dt>
          
          <code><a href="init_strict_clock_model.html">init_strict_clock_model()</a></code> 
        </dt>
        <dd>Initializes a strict clock model</dd>
      </dl><dl><dt>
          
          <code><a href="init_tn93_site_model.html">init_tn93_site_model()</a></code> 
        </dt>
        <dd>Initializes a TN93 site model</dd>
      </dl><dl><dt>
          
          <code><a href="init_tree_priors.html">init_tree_priors()</a></code> 
        </dt>
        <dd>Initializes all tree priors</dd>
      </dl><dl><dt>
          
          <code><a href="init_uniform_distr.html">init_uniform_distr()</a></code> 
        </dt>
        <dd>Initializes a uniform distribution</dd>
      </dl><dl><dt>
          
          <code><a href="init_yule_tree_prior.html">init_yule_tree_prior()</a></code> 
        </dt>
        <dd>Initializes a Yule tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="interspace.html">interspace()</a></code> 
        </dt>
        <dd>Puts spaces in between the lines</dd>
      </dl><dl><dt>
          
          <code><a href="is_alpha_param.html">is_alpha_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
alpha parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_b_pop_sizes_param.html">is_b_pop_sizes_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
b_pop_sizes parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_bd_tree_prior.html">is_bd_tree_prior()</a></code> 
        </dt>
        <dd>Determine if the object is a valid Birth Death tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_beauti_options.html">is_beauti_options()</a></code> 
        </dt>
        <dd>Determine if the object is a valid <code>beauti_options</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_beta_distr.html">is_beta_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
beta distribution,
as created by <code>create_beta_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_beta_param.html">is_beta_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
beta parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_cbs_tree_prior.html">is_cbs_tree_prior()</a></code> 
        </dt>
        <dd>Determine if the object is a valid constant coalescent Bayesian skyline prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_ccp_tree_prior.html">is_ccp_tree_prior()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
  constant coalescence population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_cep_tree_prior.html">is_cep_tree_prior()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
coalescent exponential population tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_clock_model.html">is_clock_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid clock_model</dd>
      </dl><dl><dt>
          
          <code><a href="is_clock_model_name.html">is_clock_model_name()</a></code> 
        </dt>
        <dd>Determines if the name is a valid clock model name</dd>
      </dl><dl><dt>
          
          <code><a href="is_clock_rate_param.html">is_clock_rate_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
clock_rate parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_default_mcmc.html">is_default_mcmc()</a></code> 
        </dt>
        <dd>Determine if the MCMC is a default MCMC</dd>
      </dl><dl><dt>
          
          <code><a href="is_distr.html">is_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid distribution</dd>
      </dl><dl><dt>
          
          <code><a href="is_distr_name.html">is_distr_name()</a></code> 
        </dt>
        <dd>Determines if the name is a valid distribution name</dd>
      </dl><dl><dt>
          
          <code><a href="is_exp_distr.html">is_exp_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
exponential distribution
as created by <code>create_exp_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_freq_equilibrium_name.html">is_freq_equilibrium_name()</a></code> 
        </dt>
        <dd>Checks if <code>name</code> is a valid <code>freq_equilibrium</code> argument value</dd>
      </dl><dl><dt>
          
          <code><a href="is_freq_param.html">is_freq_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid freq parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_gamma_distr.html">is_gamma_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
gamma distribution,
as created by <code>create_gamma_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_gamma_site_model.html">is_gamma_site_model()</a></code> 
        </dt>
        <dd>Is object x a gamma site model?</dd>
      </dl><dl><dt>
          
          <code><a href="is_gtr_site_model.html">is_gtr_site_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid GTR site model,
as created by <code>create_gtr_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_hky_site_model.html">is_hky_site_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid HKY site model,
as created by <code>create_hky_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_id.html">is_id()</a></code> 
        </dt>
        <dd>Determine if the object is a valid ID</dd>
      </dl><dl><dt>
          
          <code><a href="is_in_patterns.html">is_in_patterns()</a></code> 
        </dt>
        <dd>Is there at least one regular expression having a match with the line?</dd>
      </dl><dl><dt>
          
          <code><a href="is_inference_model.html">is_inference_model()</a></code> 
        </dt>
        <dd>Determine if the input is an inference model</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_bd_tree_prior.html">is_init_bd_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Birth-Death tree_prior object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_beta_distr.html">is_init_beta_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized beta distribution object
  as created by <code>create_beta_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_cbs_tree_prior.html">is_init_cbs_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Coalescent Bayesian Skyline
  tree_prior object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_ccp_tree_prior.html">is_init_ccp_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Coalescent Constant Population
  tree_prior object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_cep_tree_prior.html">is_init_cep_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Coalescent Exponential Population
  tree_prior object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_clock_model.html">is_init_clock_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized clock_model object,
as created by <code>create_clock_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_distr.html">is_init_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized distribution object
  as created by <code>create_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_exp_distr.html">is_init_exp_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized exponential distribution object
  as created by <code>create_exp_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_gamma_distr.html">is_init_gamma_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized gamma distribution object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_gamma_site_model.html">is_init_gamma_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized gamma site model,
as created by <code>create_gamma_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_gtr_site_model.html">is_init_gtr_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized GTR site model
as created by <code>create_gtr_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_hky_site_model.html">is_init_hky_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized HKY site model
as created by <code>create_hky_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_inv_gamma_distr.html">is_init_inv_gamma_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized inverse-gamma distribution
  as created by <code>create_inv_gamma_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_jc69_site_model.html">is_init_jc69_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized JC69 site model
as created by <code>create_jc69_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_laplace_distr.html">is_init_laplace_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Laplace distribution
  as created by <code>create_laplace_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_log_normal_distr.html">is_init_log_normal_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized log_normal distribution object
  as created by <code>create_log_normal_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_mrca_prior.html">is_init_mrca_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized MRCA prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_normal_distr.html">is_init_normal_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized normal distribution object
  as created by <code>create_normal_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_one_div_x_distr.html">is_init_one_div_x_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized one_div_x distribution object
  as created by <code>create_one_div_x_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_param.html">is_init_param()</a></code> 
        </dt>
        <dd>Determine if x is an initialized parameter,
  as created by create_param</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_poisson_distr.html">is_init_poisson_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Poisson distribution object
  as created by <code>create_poisson_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_rln_clock_model.html">is_init_rln_clock_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized relaxed log-normal clock_model object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_site_model.html">is_init_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized site model,
as created by <code>create_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_strict_clock_model.html">is_init_strict_clock_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized strict clock_model object</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_tn93_site_model.html">is_init_tn93_site_model()</a></code> 
        </dt>
        <dd>Determine if x is an initialized tn93 site model
as created by <code>create_tn93_site_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_tree_prior.html">is_init_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized tree_prior objects</dd>
      </dl><dl><dt>
          
          <code><a href="is_init_uniform_distr.html">is_init_uniform_distr()</a></code> 
        </dt>
        <dd>Determine if x is an initialized uniform distribution object
  as created by <code>create_uniform_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_init_yule_tree_prior.html">is_init_yule_tree_prior()</a></code> 
        </dt>
        <dd>Determine if x is an initialized Yule tree_prior object</dd>
      </dl><dl><dt>
          
          <code><a href="is_inv_gamma_distr.html">is_inv_gamma_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
inverse-gamma distribution
as created by <code>create_inv_gamma_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_jc69_site_model.html">is_jc69_site_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid JC69 site model</dd>
      </dl><dl><dt>
          
          <code><a href="is_kappa_1_param.html">is_kappa_1_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid kappa 1 parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_kappa_2_param.html">is_kappa_2_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid kappa 2 parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_kappa_param.html">is_kappa_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid kappa parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_lambda_param.html">is_lambda_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid lambda parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_laplace_distr.html">is_laplace_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
Laplace distribution,
as created by <code>create_laplace_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_log_normal_distr.html">is_log_normal_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
log-normal distribution,
as created by <code>create_log_normal_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_m_param.html">is_m_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
m parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_mcmc.html">is_mcmc()</a></code> 
        </dt>
        <dd>Determine if the object is a valid MCMC</dd>
      </dl><dl><dt>
          
          <code><a href="is_mcmc_nested_sampling.html">is_mcmc_nested_sampling()</a></code> 
        </dt>
        <dd>Determine if the object is a valid Nested-Sampling MCMC,
  as used in [1]</dd>
      </dl><dl><dt>
          
          <code><a href="is_mean_param.html">is_mean_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid mean parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_mrca_align_id_in_fasta.html">is_mrca_align_id_in_fasta()</a></code> 
        </dt>
        <dd>Determine if an MRCA prior's alignment IDs is present in the FASTA file</dd>
      </dl><dl><dt>
          
          <code><a href="is_mrca_align_ids_in_fastas.html">is_mrca_align_ids_in_fastas()</a></code> 
        </dt>
        <dd>Determine if an MRCA prior's alignment IDs are present in the FASTA files</dd>
      </dl><dl><dt>
          
          <code><a href="is_mrca_prior.html">is_mrca_prior()</a></code> 
        </dt>
        <dd>Determine of the object is an empty (<code>NA</code>) or valid MRCA prior.</dd>
      </dl><dl><dt>
          
          <code><a href="is_mrca_prior_with_distr.html">is_mrca_prior_with_distr()</a></code> 
        </dt>
        <dd>See if x is one MRCA prior with a distribution</dd>
      </dl><dl><dt>
          
          <code><a href="is_mu_param.html">is_mu_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
mu parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_normal_distr.html">is_normal_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
normal distribution
as created by <code>create_normal_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_on_appveyor.html">is_on_appveyor()</a></code> 
        </dt>
        <dd>Determines if the environment is AppVeyor</dd>
      </dl><dl><dt>
          
          <code><a href="is_on_ci.html">is_on_ci()</a></code> 
        </dt>
        <dd>Determines if the environment is a continuous integration service</dd>
      </dl><dl><dt>
          
          <code><a href="is_on_github_actions.html">is_on_github_actions()</a></code> 
        </dt>
        <dd>Determines if the environment is GitHub Actions</dd>
      </dl><dl><dt>
          
          <code><a href="is_on_travis.html">is_on_travis()</a></code> 
        </dt>
        <dd>Determines if the environment is Travis CI</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_bool.html">is_one_bool()</a></code> 
        </dt>
        <dd>Check if the argument is one boolean</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_div_x_distr.html">is_one_div_x_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
1/x distribution,
as created by <code>create_one_div_x_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_one_double.html">is_one_double()</a></code> 
        </dt>
        <dd>Determines if the argument is a double</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_empty_string.html">is_one_empty_string()</a></code> 
        </dt>
        <dd>Determine if an object is one empty string</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_int.html">is_one_int()</a></code> 
        </dt>
        <dd>Determines if the argument is a whole number</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_na.html">is_one_na()</a></code> 
        </dt>
        <dd>Determines if x is one NA</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_string.html">is_one_string()</a></code> 
        </dt>
        <dd>Determines if the argument is one string</dd>
      </dl><dl><dt>
          
          <code><a href="is_one_string_that_is_a_number.html">is_one_string_that_is_a_number()</a></code> 
        </dt>
        <dd>General function to create a distribution.</dd>
      </dl><dl><dt>
          
          <code><a href="is_param.html">is_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_param_name.html">is_param_name()</a></code> 
        </dt>
        <dd>Determines if the name is a valid parameter name</dd>
      </dl><dl><dt>
          
          <code><a href="is_phylo.html">is_phylo()</a></code> 
        </dt>
        <dd>Checks if the input is a phylogeny</dd>
      </dl><dl><dt>
          
          <code><a href="is_poisson_distr.html">is_poisson_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
Poisson distribution
as created by <code>create_poisson_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_ac_param.html">is_rate_ac_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate AC' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_ag_param.html">is_rate_ag_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate AG' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_at_param.html">is_rate_at_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate AT' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_cg_param.html">is_rate_cg_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate CG' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_ct_param.html">is_rate_ct_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate CT' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rate_gt_param.html">is_rate_gt_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
'rate GT' parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_rln_clock_model.html">is_rln_clock_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid relaxed log normal clock model</dd>
      </dl><dl><dt>
          
          <code><a href="is_s_param.html">is_s_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
s parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_scale_param.html">is_scale_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
scale parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_sigma_param.html">is_sigma_param()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
sigma parameter</dd>
      </dl><dl><dt>
          
          <code><a href="is_site_model.html">is_site_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid site_model</dd>
      </dl><dl><dt>
          
          <code><a href="is_site_model_name.html">is_site_model_name()</a></code> 
        </dt>
        <dd>Determines if the name is a valid site_model name</dd>
      </dl><dl><dt>
          
          <code><a href="is_strict_clock_model.html">is_strict_clock_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid strict clock model,
  as returned by <code>create_strict_clock_model</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_tn93_site_model.html">is_tn93_site_model()</a></code> 
        </dt>
        <dd>Determine if the object is a valid TN93 site model,</dd>
      </dl><dl><dt>
          
          <code><a href="is_tree_prior.html">is_tree_prior()</a></code> 
        </dt>
        <dd>Determine if an object is a valid tree prior</dd>
      </dl><dl><dt>
          
          <code><a href="is_tree_prior_name.html">is_tree_prior_name()</a></code> 
        </dt>
        <dd>Determines if the name is a valid tree prior name</dd>
      </dl><dl><dt>
          
          <code><a href="is_uniform_distr.html">is_uniform_distr()</a></code> 
        </dt>
        <dd>Determine if the object is a valid
uniform distribution
as created by <code>create_uniform_distr</code></dd>
      </dl><dl><dt>
          
          <code><a href="is_xml.html">is_xml()</a></code> 
        </dt>
        <dd>Checks if the text is a valid XML node, that is,
it has a opening and matching closing tag</dd>
      </dl><dl><dt>
          
          <code><a href="is_yule_tree_prior.html">is_yule_tree_prior()</a></code> 
        </dt>
        <dd>Determine if the object is a valid Yule tree prior,</dd>
      </dl><dl><dt>
          
          <code><a href="jc69_site_model_to_xml_state.html">jc69_site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="kappa_param_to_xml.html">kappa_param_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="m_param_to_xml.html">m_param_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="mcmc_to_xml_run.html">mcmc_to_xml_run()</a></code> 
        </dt>
        <dd>Converts an MCMC object to the run section's XML</dd>
      </dl><dl><dt>
          
          <code><a href="mcmc_to_xml_run_default.html">mcmc_to_xml_run_default()</a></code> 
        </dt>
        <dd>Converts an MCMC object to the run section's XML for a default MCMC</dd>
      </dl><dl><dt>
          
          <code><a href="mcmc_to_xml_run_nested_sampling.html">mcmc_to_xml_run_nested_sampling()</a></code> 
        </dt>
        <dd>Converts an MCMC object to the run section's XML for a Nested-Sampling MCMC</dd>
      </dl><dl><dt>
          
          <code><a href="mrca_prior_to_xml_prior_distr.html">mrca_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="mrca_prior_to_xml_state.html">mrca_prior_to_xml_state()</a></code> 
        </dt>
        <dd>Internal function to create the XML of an MRCA prior,
  as used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="mrca_prior_to_xml_taxonset.html">mrca_prior_to_xml_taxonset()</a></code> 
        </dt>
        <dd>Creates the <code>taxonset</code> section in the prior section of the
distribution section of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="mrca_prior_to_xml_tracelog.html">mrca_prior_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="mrca_priors_to_xml_prior_distr.html">mrca_priors_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the the <code>distribution</code>'s prior section (which is part of
a posterior distribution section) of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="no_taxa_to_xml_tree.html">no_taxa_to_xml_tree()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml.html">parameter_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_kappa_1.html">parameter_to_xml_kappa_1()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_kappa_2.html">parameter_to_xml_kappa_2()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_lambda.html">parameter_to_xml_lambda()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_mean.html">parameter_to_xml_mean()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_mu.html">parameter_to_xml_mu()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_ac.html">parameter_to_xml_rate_ac()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_ag.html">parameter_to_xml_rate_ag()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_at.html">parameter_to_xml_rate_at()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_cg.html">parameter_to_xml_rate_cg()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_ct.html">parameter_to_xml_rate_ct()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_rate_gt.html">parameter_to_xml_rate_gt()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_scale.html">parameter_to_xml_scale()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="parameter_to_xml_sigma.html">parameter_to_xml_sigma()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="remove_beautier_folder.html">remove_beautier_folder()</a></code> 
        </dt>
        <dd>Check there are no files in the default beautier folder</dd>
      </dl><dl><dt>
          
          <code><a href="remove_empty_lines.html">remove_empty_lines()</a></code> 
        </dt>
        <dd>Remove all lines that are only whitespace</dd>
      </dl><dl><dt>
          
          <code><a href="remove_multiline.html">remove_multiline()</a></code> 
        </dt>
        <dd>Remove consecutive lines</dd>
      </dl><dl><dt>
          
          <code><a href="rename_inference_model_filenames.html">rename_inference_model_filenames()</a></code> 
        </dt>
        <dd>Rename the filenames in an inference model</dd>
      </dl><dl><dt>
          
          <code><a href="rename_mcmc_filenames.html">rename_mcmc_filenames()</a></code> 
        </dt>
        <dd>Rename the filenames within an MCMC</dd>
      </dl><dl><dt>
          
          <code><a href="rln_clock_model_to_xml_mean_rate_prior.html">rln_clock_model_to_xml_mean_rate_prior()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="rln_clock_model_to_xml_operators.html">rln_clock_model_to_xml_operators()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="rln_clock_model_to_xml_prior_distr.html">rln_clock_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="rln_clock_model_to_xml_state.html">rln_clock_model_to_xml_state()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="rln_clock_model_to_xml_tracelog.html">rln_clock_model_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="rnd_phylo_to_xml_init.html">rnd_phylo_to_xml_init()</a></code> 
        </dt>
        <dd>Creates the XML of a random phylogeny,
  as used in the <code>init</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="s_parameter_to_xml.html">s_parameter_to_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="site_model_to_xml_operators.html">site_model_to_xml_operators()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>operators</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="site_model_to_xml_prior_distr.html">site_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="site_model_to_xml_state.html">site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Internal function to convert a site model to XML,
  used in the `state` section</dd>
      </dl><dl><dt>
          
          <code><a href="site_model_to_xml_tracelog.html">site_model_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Creates the site model's XML for the tracelog section</dd>
      </dl><dl><dt>
          
          <code><a href="site_models_to_xml_operators.html">site_models_to_xml_operators()</a></code> 
        </dt>
        <dd>Write the XML <code>operators</code> section from the site models.</dd>
      </dl><dl><dt>
          
          <code><a href="site_models_to_xml_prior_distr.html">site_models_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Represent the site models as XML</dd>
      </dl><dl><dt>
          
          <code><a href="site_models_to_xml_tracelog.html">site_models_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Creates the site models' XML for the tracelog section</dd>
      </dl><dl><dt>
          
          <code><a href="strict_clock_model_to_xml_operators.html">strict_clock_model_to_xml_operators()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="strict_clock_model_to_xml_prior_distr.html">strict_clock_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="strict_clock_model_to_xml_state.html">strict_clock_model_to_xml_state()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="strict_clock_model_to_xml_tracelog.html">strict_clock_model_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="taxa_to_xml_tree.html">taxa_to_xml_tree()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tipdate_taxa_to_xml_trait.html">tipdate_taxa_to_xml_trait()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tipdate_taxa_to_xml_tree.html">tipdate_taxa_to_xml_tree()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tn93_site_model_to_xml_prior_distr.html">tn93_site_model_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tn93_site_model_to_xml_state.html">tn93_site_model_to_xml_state()</a></code> 
        </dt>
        <dd>Converts a site model to XML,
  used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="tree_model_to_tracelog_xml.html">tree_model_to_tracelog_xml()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tree_prior_to_xml_operators.html">tree_prior_to_xml_operators()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="tree_prior_to_xml_prior_distr.html">tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="tree_prior_to_xml_state.html">tree_prior_to_xml_state()</a></code> 
        </dt>
        <dd>Creates the XML of a tree prior,
  as used in the <code>state</code> section</dd>
      </dl><dl><dt>
          
          <code><a href="tree_prior_to_xml_tracelog.html">tree_prior_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Creates the tree prior's XML for the tracelog section</dd>
      </dl><dl><dt>
          
          <code><a href="tree_priors_to_xml_prior_distr.html">tree_priors_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the distribution section in the prior section of the
distribution section of a BEAST2 XML parameter file.</dd>
      </dl><dl><dt>
          
          <code><a href="tree_priors_to_xml_tracelog.html">tree_priors_to_xml_tracelog()</a></code> 
        </dt>
        <dd>Creates the tree priors' XML for the tracelog section</dd>
      </dl><dl><dt>
          
          <code><a href="unindent.html">unindent()</a></code> 
        </dt>
        <dd>Unindents text</dd>
      </dl><dl><dt>
          
          <code><a href="yule_tree_prior_to_xml_operators.html">yule_tree_prior_to_xml_operators()</a></code> 
        </dt>
        <dd>Internal function</dd>
      </dl><dl><dt>
          
          <code><a href="yule_tree_prior_to_xml_prior_distr.html">yule_tree_prior_to_xml_prior_distr()</a></code> 
        </dt>
        <dd>Creates the <code>prior</code> section in the prior section of
the prior section of the distribution section
of a BEAST2 XML parameter file for a Yule tree prior</dd>
      </dl></div>
  </main></div>


    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg"></div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul><h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul></div>
                    <div class="col-md-3 col-sm-4">
                        <ul><h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul></div>
                    <div class="col-md-2 col-sm-4">
                        <ul><h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul></div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer></div>

  

  

  </body></html>

