<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="beautier">
<title>Inference Models • beautier</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Inference Models">
<meta property="og:description" content="beautier">
<meta property="og:image" content="https://docs.ropensci.org/beautier/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">beautier</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.6.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/demo.html">beautier demo</a>
    <a class="dropdown-item" href="../articles/examples.html">Examples</a>
    <a class="dropdown-item" href="../articles/inference_models.html">Inference Models</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/beautier/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Inference Models</h1>
                        <h4 data-toc-skip class="author">Richèl J.C. Bilderbeek</h4>
            
            <h4 data-toc-skip class="date">2024-03-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/beautier/blob/HEAD/vignettes/inference_models.Rmd" class="external-link"><code>vignettes/inference_models.Rmd</code></a></small>
      <div class="d-none name"><code>inference_models.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><img src="beautier_logo.png"></p>
<p><code>beautier</code> allows to specify an inference model, from which, together with a DNA alignment, a posterior (holding a distribution of phylogenies and jointly-estimated parameter estimates) can be inferred.</p>
<p>The inference model entails the evolutionary model used in inference, as well as some settings to do the inference.</p>
<p>This vignette shows the options how to specify an inference model, and which options are possible.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>First, load <code>beautier</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/beautier/">beautier</a></span><span class="op">)</span></span></code></pre></div>
<p>Now, we’ll look at the default inference model to get an idea of what an inference model entails.</p>
</div>
<div class="section level2">
<h2 id="default-inference-model">Default inference model<a class="anchor" aria-label="anchor" href="#default-inference-model"></a>
</h2>
<p>Creating a default inference model is easy:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>An inference model is a list, that uses the BEAST2 defaults. Here are the elements of that list:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">inference_model</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "site_model"        "clock_model"       "tree_prior"       </span></span>
<span><span class="co">#&gt; [4] "mrca_prior"        "mcmc"              "beauti_options"   </span></span>
<span><span class="co">#&gt; [7] "tipdates_filename"</span></span></code></pre></div>
<p>As we can see, an inference model entails these elements, which we will cover below in more detail:</p>
<ul>
<li>a site model: how the alignment changes</li>
<li>a clock model: how the mutation rates differ over the branches</li>
<li>a tree prior: the speciation model</li>
<li>(optional) an MRCA prior: a ‘Most Recent Common Ancestor’</li>
<li>an MCMC: the Markov Chain Monte Carlo setup</li>
<li>(optional) BEAUti options: version-specific options</li>
<li>(experimental) a tipdates filename: a filename for tip-dating</li>
</ul>
<div class="section level3">
<h3 id="site-models">Site models<a class="anchor" aria-label="anchor" href="#site-models"></a>
</h3>
<p>The site model entails how the alignment changes over time. Currently, <code>beautier</code> supplies a gamma site model. One element of a site model, for DNA/RNA, is the nucleotide substitution model (‘NSM’).</p>
<p>Due to historical reasons, <code>beautier</code> confuses the site model and NSM: <code>beautier</code> has no functions with the word ‘nucleotide substitution’ (nor <code>nsm</code>) `in it. Instead, it is as if these are specific site models.</p>
<p>To see the available site models, use <code><a href="../reference/create_site_model.html">?create_site_model</a></code> to see a list, or use:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_site_model_names.html">get_site_model_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "JC69" "HKY"  "TN93" "GTR"</span></span></code></pre></div>
<p>The simplest NSM is the JC69 NSM, which assumes all nucleotides are substituted by one another at the same rate. As an example, to use a gamma site model with the JC69 NSM model in inference:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  site_model <span class="op">=</span> <span class="fu"><a href="../reference/create_jc69_site_model.html">create_jc69_site_model</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clock-models">Clock models<a class="anchor" aria-label="anchor" href="#clock-models"></a>
</h3>
<p>The clock model entails how the mutation rates differ over the branches.</p>
<p>To see the available site models, use <code><a href="../reference/create_clock_model.html">?create_clock_model</a></code> to see a list, or use:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_clock_model_names.html">get_clock_model_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "relaxed_log_normal" "strict"</span></span></code></pre></div>
<p>The simplest clock model is the strict clock model, which assumes all branches have the same mutation rate. As an example, to use a strict clock model in inference:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  clock_model <span class="op">=</span> <span class="fu"><a href="../reference/create_strict_clock_model.html">create_strict_clock_model</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="tree-prior">Tree prior<a class="anchor" aria-label="anchor" href="#tree-prior"></a>
</h2>
<p>The tree prior is the tree model used in inference. It is called ‘tree prior’ instead of ‘tree model’, as this follow the BEAUti naming. The tree model specifies the branching process of a tree.</p>
<p>To see the available tree models, use <code><a href="../reference/create_tree_prior.html">?create_tree_prior</a></code> to see a list, or use:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_prior_names.html">get_tree_prior_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "birth_death"                    "coalescent_bayesian_skyline"   </span></span>
<span><span class="co">#&gt; [3] "coalescent_constant_population" "coalescent_exp_population"     </span></span>
<span><span class="co">#&gt; [5] "yule"</span></span></code></pre></div>
<p>The simplest tree model is the Yule (aka pure-birth) tree model, which assumes that branching events occur at a constant rate, and there are no extinctions. As an example, to use a Yule tree model in inference:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  tree_prior <span class="op">=</span> <span class="fu"><a href="../reference/create_yule_tree_prior.html">create_yule_tree_prior</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="optional-mrca-prior-a-most-recent-common-ancestor">(optional) MRCA prior: a ‘Most Recent Common Ancestor’<a class="anchor" aria-label="anchor" href="#optional-mrca-prior-a-most-recent-common-ancestor"></a>
</h2>
<p>With the MRCA (‘Most Recent Common Ancestor’) prior, one can specify which tips share a common ancestor.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The alignmet</span></span>
<span><span class="va">fasta_filename</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_beautier_path.html">get_beautier_path</a></span><span class="op">(</span><span class="st">"anthus_aco.fas"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The alignment's ID</span></span>
<span><span class="va">alignment_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_alignment_id.html">get_alignment_id</a></span><span class="op">(</span><span class="va">fasta_filename</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get the first two taxa's names</span></span>
<span><span class="va">taxa_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_taxa_names.html">get_taxa_names</a></span><span class="op">(</span><span class="va">fasta_filename</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Specify that the first two taxa share a common ancestor</span></span>
<span><span class="va">mrca_prior</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_mrca_prior.html">create_mrca_prior</a></span><span class="op">(</span></span>
<span>  alignment_id <span class="op">=</span> <span class="va">alignment_id</span>,</span>
<span>  taxa_names <span class="op">=</span> <span class="va">taxa_names</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use the MRCA prior in inference</span></span>
<span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  mrca_prior <span class="op">=</span> <span class="va">mrca_prior</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="mcmc-the-markov-chain-monte-carlo-setup">MCMC: the Markov Chain Monte Carlo setup<a class="anchor" aria-label="anchor" href="#mcmc-the-markov-chain-monte-carlo-setup"></a>
</h2>
<p>The MCMC (‘Markov Chain Monte Carlo’) specifies how the inference algorithm does its work.</p>
<p>The available MCMC’s can be found using <code><a href="../reference/create_mcmc.html">?create_mcmc</a></code> and are:</p>
<ul>
<li>
<code>create_mcmc</code>: regular MCMC</li>
<li>
<code>create_test_mcmc</code>: shorter regular MCMC, to be used in testing</li>
<li>
<code>create_ns_mcmc</code>: MCMC to estimate a marginal likelihood using nested sampling</li>
</ul>
</div>
<div class="section level2">
<h2 id="optional-beauti-options-version-specific-options">(optional) BEAUti options: version-specific options<a class="anchor" aria-label="anchor" href="#optional-beauti-options-version-specific-options"></a>
</h2>
<p>The BEAUti options entail version-specific options to store an inference model as a BEAST2 XML input file.</p>
<p>The available BEAUti options can be found using <code><a href="../reference/create_beauti_options.html">?create_beauti_options</a></code> and are:</p>
<ul>
<li>
<code>create_beauti_options_v2_4</code>: BEAUti v2.4</li>
<li>
<code>create_beauti_options_v2_6</code>: BEAUti v2.6</li>
</ul>
<p>Using a specific version for an inference:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  beauti_options <span class="op">=</span> <span class="fu"><a href="../reference/create_beauti_options_v2_4.html">create_beauti_options_v2_4</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="experimental-a-tipdates-filename-a-filename-for-tip-dating">(experimental) a tipdates filename: a filename for tip-dating<a class="anchor" aria-label="anchor" href="#experimental-a-tipdates-filename-a-filename-for-tip-dating"></a>
</h2>
<p>A tipdates filename is an experimental feature for tip-dating:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_inference_model.html">create_inference_model</a></span><span class="op">(</span></span>
<span>  tipdates_filename <span class="op">=</span> <span class="fu"><a href="../reference/get_beautier_path.html">get_beautier_path</a></span><span class="op">(</span><span class="st">"G_VII_pre2003_dates_4.txt"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The tipdates filename and the alignment must be compatible. Here is an example:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output_filename</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_beautier_tempfilename.html">get_beautier_tempfilename</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/create_beast2_input_file_from_model.html">create_beast2_input_file_from_model</a></span><span class="op">(</span></span>
<span>  input_filename <span class="op">=</span> <span class="fu"><a href="../reference/get_beautier_path.html">get_beautier_path</a></span><span class="op">(</span><span class="st">"G_VII_pre2003_msa.fas"</span><span class="op">)</span>,</span>
<span>  inference_model <span class="op">=</span> <span class="va">inference_model</span>,</span>
<span>  output_filename <span class="op">=</span> <span class="va">output_filename</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Cleanup</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">output_filename</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span></span>
<span><span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="../reference/remove_beautier_folder.html">remove_beautier_folder</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">beautier</span><span class="fu">::</span><span class="fu"><a href="../reference/check_empty_beautier_folder.html">check_empty_beautier_folder</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
